components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -378,9 +378,10 @@ public Device getDevice(DeviceIdentifier deviceId) throws DeviceManagementExcept
     }
 
     @Override
-    public boolean updateDeviceInfo(Device device) throws DeviceManagementException {
-        DeviceManager dms = this.getPluginRepository().getDeviceManagementProvider(device.getType());
-        return dms.updateDeviceInfo(device);
+    public boolean updateDeviceInfo(DeviceIdentifier deviceIdentifier, Device device) throws DeviceManagementException {
```

Comment:
In AndroidDeviceDAO, it is implemented to return true when a device successfully add, remove or update. Hence it is being used here as well to acknowledge.